src/session/user.rs

```diff
@@ -146,12 +147,64 @@ mod imp {
 
             let avatar = obj.avatar();
             let user_expression = gtk::ConstantExpression::new(obj);
-            let full_name_expression = super::User::full_name_expression(&user_expression);
-            full_name_expression.bind(avatar, "display-name", gtk::NONE_WIDGET);
+            let display_name_expression = super::User::display_name_expression(&user_expression);
+            display_name_expression.bind(avatar, "display-name", gtk::NONE_WIDGET);
         }
     }
 }
 
+fn last_seen_time_ago(unix: i32) -> String {
+    let datetime_now = glib::DateTime::new_now_local().unwrap();
+    let last_seen = glib::DateTime::from_unix_utc(unix.into())
+        .unwrap()
+        .to_local()
+        .unwrap();
+
+    let seconds = datetime_now.difference(&last_seen) / 1_000_000;
+    let minutes = seconds / 60;
+    let hours = minutes / 60;
+    let days = hours / 24;
+
+    if minutes == 0 {
+        gettext("last seen just now")
+    } else if minutes == 1 {
+        gettext("last seen a minute ago")
+    } else if minutes < 60 {
+        gettext!("last seen {} minutes ago", minutes)
+    } else if minutes < 120 {
+        gettext("last seen an hour ago")
+    } else if (datetime_now.day_of_month() == last_seen.day_of_month()) & (hours < 24) {
```

Comment:
Why is it important that this has to be on the same day?
I mean, we want to print a duration instead of an absolute point in time. As a user, I want to know whether my contact was online 5 hours ago. I couldn't care less if 5 hours ago was on the previous day. I still want to see `last seen 5 hours ago`

```suggestion
    } else if hours < 24 {
```

Reply:
@just-carlod Does the official client handle it in the same way?

Reply:
> As a user, I want to know whether my contact was online 5 hours ago. I couldn't care less if 5 hours ago was on the previous day. I still want to see `last seen 5 hours ago`

I guess we have different needs then. We can always add more ways of time counting in the future, so users will decide which one they want to use

Reply:
Like I said in the 2nd comment, let's do it the way as you originally committed it. I don't know what's @melix99 opinion, but for me, it's ok as it seems the official client does it in the same way.

> We can always add more ways of time counting in the future, so users will decide which one they want to use

A big NO from me here :-P. Let's not overcomplicate things.

Reply:
> `hours < 24` are required to determine whether or not it's the same month and year. It's more readable than `(datetime.year(), datetime.month()) == (last_seen.year(), last_seen.month())`

Oh, my bad. Did not look close enough. :-) Then just change this.
```suggestion
    } else if datetime_now.day_of_month() == last_seen.day_of_month() && hours < 24 {
```